src/navigation/RootNavigator.tsx

```diff
@@ -3,7 +3,7 @@ import { FC } from 'react'
 
 import { BottomTabNavigator } from './BottomTabNavigator'
 
-import { useAuth } from '~hooks'
+import { useAuth } from '~providers'
```

Comment:
Our code structure has programmer expect to have hooks within `~hooks` directory. I think that you could use `createGenericContext` within `context` directory and reexport `AuthProvider` in `providers` dir and `useAuth` in `hooks` dir.